src/main/kotlin/com/bugsnag/android/gradle/internal/GradleUtil.kt

```diff
@@ -0,0 +1,92 @@
+@file:Suppress("MatchingDeclarationName") // This file contains multiple top-level members
+package com.bugsnag.android.gradle.internal
+
+import okio.HashingSource
+import okio.blackholeSink
+import okio.buffer
+import okio.source
+import org.gradle.api.Task
+import org.gradle.api.invocation.Gradle
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.MapProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.SetProperty
+import org.gradle.api.tasks.TaskContainer
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.util.VersionNumber
+import java.io.File
+
+internal object GradleVersions {
+    val VERSION_5_3: VersionNumber = VersionNumber.parse("5.3")
+    val VERSION_6: VersionNumber = VersionNumber.parse("6.0")
+    val VERSION_6_1: VersionNumber = VersionNumber.parse("6.1")
+    val VERSION_6_6: VersionNumber = VersionNumber.parse("6.6")
+}
+
+internal fun Gradle.versionNumber(): VersionNumber = VersionNumber.parse(gradleVersion)
+
+/** A fast file hash that don't load the entire file contents into memory at once. */
+internal fun File.md5HashCode(): Int {
+    return HashingSource.md5(source()).use { hashingSource ->
+        hashingSource.buffer().readAll(blackholeSink())
+        hashingSource.hash.hashCode()
+    }
+}
+
+/* Borrowed helper functions from the Gradle Kotlin DSL. */
```

Comment:
I believe you can just use kotlin-dsl directly without imposing it on your downstream consumers, but this subset of APIs was easy enough to just copy in for now

Reply:
Looks good :) if extra functionality is required in future it'd be worth setting up the kotlin-dsl